src/post/post.service.ts

```diff
@@ -33,8 +33,23 @@ export class PostService {
     return `This action returns a #${id} post`;
   }
 
-  update(id: number, updatePostDto: UpdatePostDto) {
-    return `This action updates a #${id} post`;
+  async update(id: string, updatePostDto: UpdatePostDto): Promise<ResponsePostDTO> { 
+    const existingPost = await this.postModel.findByIdAndUpdate(id, updatePostDto, { new: true });
+    
+    if (!existingPost) {
+      throw new NotFoundException(`Post with id ${id} not found`);
+    }
+
+    const updatedPost = {
+      id: existingPost.id,
+      userId: existingPost.userId,
+      content: existingPost.content,
+      mediaUrl: existingPost.mediaUrl,
+      likes: existingPost.likes,
+      status: existingPost.status
+    };
+
+    return updatedPost;
```

Comment:
Can you return existingPost instead of updatePost?

Reply:
No, because when i return exitstingPost, It will include all the info unnecessary like createAt, updateAt,.... So i should create a updatedPost to return information necessary

Reply:
Since user dont need the data of resource after update to render so, you dont need return the post data. Only need status.